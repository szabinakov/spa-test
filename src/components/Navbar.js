import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUserPlus, faHome } from "@fortawesome/free-solid-svg-icons";
import "../styles/Navbar.css";

export default function Navbar({ userState }) {
  const isUserLoggedIn = userState.isUserLoggedIn;

  return (
    <div data-testid="navContainer" className="navContainer">
      {isUserLoggedIn && (
        <ul>
          <li data-testid="homeLink" className="homeLink">
            <Link to="/">
              <FontAwesomeIcon className="iconContainer" icon={faHome} />
            </Link>
          </li>
          <li data-testid="addUserLink" className="addUserLink">
            <Link to="/add-users">
              <FontAwesomeIcon className="iconContainer" icon={faUserPlus} />
            </Link>
          </li>
        </ul>
      )}
    </div>
  );
}

Navbar.propTypes = {
  userState: PropTypes.object,
};
