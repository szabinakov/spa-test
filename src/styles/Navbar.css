.navContainer > ul {
  display: flex;
  width: 200px;
  margin-top: 0px;
  padding-left: 5px;
  justify-content: space-evenly;
}

.navContainer > ul > li:hover {
  border-bottom: 2px solid var(--orange);
}

.navContainer > ul > li {
  display: flex;
  justify-content: center;
  list-style: none;
  width: 70px;
}
.navContainer > ul > li > a {
  text-decoration: none;
  color: var(--darkblue);
  font-size: 30px;
}
