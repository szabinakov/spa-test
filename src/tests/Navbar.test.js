import { render } from "@testing-library/react";
import Navbar from "../components/Navbar.js";
import { BrowserRouter as Router } from "react-router-dom";

describe("Navbar", () => {
  const mockStateTrue = {
    isUserLoggedIn: true,
  };
  const mockStateFalse = {
    isUserLoggedIn: false,
  };

  it("Matches the snapshot when user is logged in", () => {
    const { asFragment } = render(
      <Router>
        <Navbar userState={mockStateTrue} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("it renders every element when user is logged in", () => {
    const { getByTestId } = render(
      <Router>
        <Navbar userState={mockStateTrue} />
      </Router>
    );

    expect(getByTestId("homeLink")).toHaveClass("homeLink");
    expect(getByTestId("addUserLink")).toHaveClass("addUserLink");
  });

  it("does not matches the snapshot when user is not logged in", () => {
    const { asFragment } = render(
      <Router>
        <Navbar userState={mockStateFalse} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("it doesnt render links when user is not logged in ", () => {
    const { getByTestId } = render(
      <Router>
        <Navbar userState={mockStateFalse} />
      </Router>
    );
    expect(getByTestId("navContainer")).toBeEmptyDOMElement();
  });
});
