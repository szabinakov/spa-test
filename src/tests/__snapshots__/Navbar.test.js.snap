// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Matches the snapshot when user is logged in 1`] = `
<DocumentFragment>
  <div
    class="navContainer"
    data-testid="navContainer"
  >
    <ul>
      <li
        class="homeLink"
        data-testid="homeLink"
      >
        <a
          href="/"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-home fa-w-18 iconContainer"
            data-icon="home"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
      <li
        class="addUserLink"
        data-testid="addUserLink"
      >
        <a
          href="/add-users"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user-plus fa-w-20 iconContainer"
            data-icon="user-plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Navbar does not matches the snapshot when user is not logged in 1`] = `
<DocumentFragment>
  <div
    class="navContainer"
    data-testid="navContainer"
  />
</DocumentFragment>
`;
